docs/source/dataframe.rst

```diff
@@ -12,6 +12,7 @@ Dask DataFrame
    Internal Design <dataframe-design.rst>
    Best Practices <dataframe-best-practices.rst>
    API <dataframe-api.rst>
+   API Query Planning <dask-expr-api.rst>
```

Comment:
My sense is this is too top-level at the moment. I'd recommend moving this into the "Additional Information" dropdown for now unless there's a specific reason it needs to be top-level.

If it does need to be top-level, then we'll probably want more information about the query planning effort more generally that answers questions like "What is it?", "Why should I care?", "How do I enable it?", etc.

Reply:
Yeah I'll move it down, but we ideally want to add the deprecation message soonish, would move it back up then, but we will hopefully have more content by then